src/contest/weglide/UploadIGCFile.cpp

```diff
@@ -0,0 +1,207 @@
+/*
+  Copyright_License {
+
+  XCSoar Glide Computer - http://www.xcsoar.org/
+  Copyright (C) 2000-2022 The XCSoar Project
+  A detailed list of copyright holders can be found in the file "AUTHORS".
+
+  This program is free software; you can redistribute it and/or
+  modify it under the terms of the GNU General Public License
+  as published by the Free Software Foundation; either version 2
+  of the License, or (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+}
+*/
+
+#include "UploadIGCFile.hpp"
+#include "LocalPath.hpp"
+#include "Interface.hpp"
+#include "UIGlobals.hpp"
+#include "LogFile.hpp"
+#include "Cloud/weglide/UploadFlight.hpp"
+#include "Cloud/weglide/WeGlideSettings.hpp"
+#include "co/InvokeTask.hxx"
+#include "Dialogs/Message.hpp"
+#include "Dialogs/CoDialog.hpp"
+#include "Formatter/TimeFormatter.hpp"
+#include "json/ParserOutputStream.hxx"
+#include "Language/Language.hpp"
+#include "net/http/Init.hpp"
+#include "Operation/PluggableOperationEnvironment.hpp"
+#include "system/ConvertPathName.hpp"
+#include "system/FileUtil.hpp"
+#include "util/StaticString.hxx"
+#include "util/ConvertString.hpp"
+
+#include <cinttypes>
+
+/**
+ * Wrapper for getting string values of a json value
+ *
+ * @param value - the json object value
```

Comment:
Why does this explanation begin with a dash?

Reply:
To split the explanation from the param (my previous coding style)
removed because unused anymore